src/transformers/bettertransformers/bert/modeling_bert_fast.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2022 The HuggingFace and Meta Team.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import torch
+import torch.nn as nn
+
+
+class BertLayerFast(nn.Module):
```

Comment:
IIUC, this is a kind of conversion script from our BERT model to the Transformer supported by BetterTransformers?
So this could live in `modeling_bert.py` as a method similar to `load_tf_weights_in_bert`? I'd favor that instead of having a separate module.

The rest seems really cool! I like how it doesn't seem to change anything to the modeling file, seeing as there's only a conversion script.

Reply:
Great thanks a lot for the feedback! 💪 

Actually the whole conversion magic happens in the parent class, more precisely [here](https://github.com/younesbelkada/transformers/blob/fdb848767b9ab87ffaa2e042d19207ef1d5da015/src/transformers/modeling_utils.py#L2367) so that any child class from a `PretrainedModel` class can benefit from the conversion to its `BetterTransformers` equivalent - and will properly yield to an error if the `BetterTransformer` version of the model has not been implemented.

The idea here is to replace all `xxxLayer` modules to its `xxxLayerFast` module according to [this dictionary](https://github.com/younesbelkada/transformers/blob/fdb848767b9ab87ffaa2e042d19207ef1d5da015/src/transformers/bettertransformers/__init__.py#L19) - this way supporting `BetterTransformers` for models that have copied `BertLayer` will just require to update this dictionary.

Another reason why I would advocate to use `Fast` modules is because some modifications have to be done during the forward pass (e.g. in `BetterTransformers` the `attention_mask` needs to be used only by the first layer that transform the `hidden_states` into a nested padded tensor) , otherwise I believe that we'll have to change the modeling script for eg like [here](https://github.com/HamidShojanazeri/transformers/blob/ddf0299a13e7c4f54459a0731abd80204a1078f5/src/transformers/models/bert/modeling_bert.py#L640)

Again thanks for the feedback ;)